pkg/nats/stream/producer_client.go

```diff
@@ -111,17 +111,19 @@ func (pc *ProducerClient) heartBeat(ctx context.Context) {
 			pc.mu.RLock()
 
 			for c, v := range pc.activeConnHeartBeatRequestSubjects {
-				// Create an empty slice for activeStreamIds
-				activeStreamIds := make(map[string][]string)
+				// Create an empty slice for activeStreamIdsByReqSubj
+				activeStreamIdsByReqSubj := make(map[string][]string)
+				var activeStreamIds []string
 
 				if streamInfoMap, ok := pc.activeStreamInfo[c]; ok {
-					for _, streamInfo := range streamInfoMap {
-						activeStreamIds[streamInfo.RequestSub] = append(activeStreamIds[streamInfo.RequestSub], streamInfo.ID)
+					for streamId, streamInfo := range streamInfoMap {
+						activeStreamIds = append(activeStreamIds, streamId)
+						activeStreamIdsByReqSubj[streamInfo.RequestSub] = append(activeStreamIdsByReqSubj[streamInfo.RequestSub], streamInfo.ID)
```

Comment:
is `streamInfo.ID` the same as `streamId`? 

There are too many maps in `ProducerClient`, `activeStreamInfo` and `activeConnHeartBeatRequestSubjects`, but still they are not enough as we have to construct this `activeStreamIdsByReqSubj` with every heartbeat. Is there anything we can do to make things more readable? If having two maps already are not making things more readable or efficient as we still have to loop here, does it make sense to just have one `activeStreamInfo` type and have helper methods that return grouped by streams on-demand?

Reply:
Let me revisit it !

Reply:
Made it more readable!